Lib/fontTools/varLib/instancer.py

```diff
@@ -1099,6 +1117,379 @@ def pruningUnusedNames(varfont):
+    nonRibbiNameIDs = [n for n in axisValueNameIDs if n not in ribbiNameIDs]
+    elidedNameID = stat.ElidedFallbackNameID
+    elidedNameIsRibbi = nameIDIsRibbi(nametable, elidedNameID)
+
+    getName = nametable.getName
+    platforms = set((r.platformID, r.platEncID, r.langID) for r in nametable.names)
+    for platform in platforms:
+        if not all(getName(i, *platform) for i in (1, 2, elidedNameID)):
+            # Since no family name and subfamily name records were found,
+            # we cannot update this set of name Records.
+            continue
+
+        subFamilyName = " ".join(
+            getName(n, *platform).toUnicode() for n in ribbiNameIDs
+        )
+        typoSubFamilyName = " ".join(
+            getName(n, *platform).toUnicode()
+            for n in axisValueNameIDs
+            if nonRibbiNameIDs
+        )
+
+        # If neither subFamilyName and typographic SubFamilyName exist,
+        # we will use the STAT's elidedFallbackName
+        if not typoSubFamilyName and not subFamilyName:
+            if elidedNameIsRibbi:
+                subFamilyName = getName(elidedNameID, *platform).toUnicode()
+            else:
+                typoSubFamilyName = getName(elidedNameID, *platform).toUnicode()
+
+        familyNameSuffix = " ".join(
+            getName(n, *platform).toUnicode() for n in nonRibbiNameIDs
+        )
+
+        _updateNameTableStyleRecords(
+            varfont,
+            familyNameSuffix,
+            subFamilyName,
+            typoSubFamilyName,
+            *platform,
+        )
+
+
+def nameIDIsRibbi(nametable, nameID):
+    engNameRecords = any(
+        r
+        for r in nametable.names
+        if (r.platformID, r.platEncID, r.langID) == (3, 1, 0x409)
+    )
+    if not engNameRecords:
+        raise ValueError(
+            f"Canot determine if there are RIBBI Axis Value Tables "
+            "since there are no name table Records which have "
+            "platformID=3, platEncID=1, langID=0x409"
+        )
+    return (
+        True
+        if nametable.getName(nameID, 3, 1, 0x409).toUnicode()
+        in ("Regular", "Italic", "Bold", "Bold Italic")
+        else False
+    )
+
+
+def _updateNameTableStyleRecords(
+    varfont,
+    familyNameSuffix,
+    subFamilyName,
+    typoSubFamilyName,
+    platformID=3,
+    platEncID=1,
+    langID=0x409,
+):
+    # TODO (Marc F) It may be nice to make this part a standalone
+    # font renamer in the future.
+    nametable = varfont["name"]
+    platform = (platformID, platEncID, langID)
+
+    currentFamilyName = nametable.getName(
+        NameID.TYPOGRAPHIC_FAMILY_NAME, *platform
+    ) or nametable.getName(NameID.FAMILY_NAME, *platform)
+
+    currentStyleName = nametable.getName(
+        NameID.TYPOGRAPHIC_SUBFAMILY_NAME, *platform
+    ) or nametable.getName(NameID.SUBFAMILY_NAME, *platform)
+
+    currentFamilyName = currentFamilyName.toUnicode()
+    currentStyleName = currentStyleName.toUnicode()
```

Comment:
if neither typo nor legacy family/style name ids are present, calling `toUnicode()` on a None value will raise TypeError. I think getName returns None if the requested name id is missing. You should check and handle that case. I know it's unlikely that the basic two faminy/style names are missing from a valid name table, but you never know what you're thrown at